### PR TITLE
The beginning of the Journey

### PR DESCRIPTION
**Pull request by cs-szabolcs**
Sorry for bundling all of these together in a single commit.

For those who have seen the previous version: I removed the circumstances because, though it would have been great to have some influence on the later scenes, it prevents pregeneration.

Description:

Every layout corresponds to a region. Within every layout, there is a fixed number of areas. For every area, there is a scene. Currently, there are five types of scenes (outside of the intro and outro scenes): Battle, FinalBattle, Riddle, Rest, Treasure.

 For every layout a single mission is generated. Currently there are 2 types of missions:
 
 - Target: Kill someone
 _ Artifact: Retrieve something
 
The scenes within the layout are almost random, with a few exceptions:
 
 - The first scene is always an intro
 - The last scene is always an outro
 - The last 3 scenes (not counting the outro scene) are always RestScene -> BattleScene -> FinalBattleScene
 - Every other scene is selected randomly from TreasureScene, RiddleScene, BattleScene
 
 The whole Journey class can be saved to a dictionary (json) and loaded from it. It is only for test purposes.

Note: Generations can and should be fine-tuned. For example the solution for the riddle tend to be always "echo" for some reasons.

### CODE REVIEW

File: src/Generation/Descriptions/area/init/model.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from pydantic import BaseModel, Field, conlist
+
+# It is necessary to the generate both the description and the name at the same time. Though it's probably not the nicest solution
```
Comment (bensku) [inline src/Generation/Descriptions/area/init/model.py:3]: I can think of several alternatives:
1. Have a name and description as separate fields in model, not lists; ask for more than one generation. This is slow and consumes quite many tokens.
2. Generate a list of names. Then, for each of them, generate a description. I have noticed that this tends to improve the quality of names generated, but it is somewhat annoying to implement. Especially so, if performance is a concern.

Though if the current solution _works_, I don't think it needs to be changed.

File: src/Generation/Descriptions/summarize_context/model.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from pydantic import BaseModel, Field
+
+MAX_LENGTH = 300
+
+class Model(BaseModel):
+    summarized_context: str = Field(title=f'The summarized context at most {MAX_LENGTH} characters', max_length=MAX_LENGTH)
```
Comment (bensku) [inline src/Generation/Descriptions/summarize_context/model.py]: Most LLMs see text as words, not characters - additionally, they are quite bad at counting. If this works, no need to change, but if not, referring to words/paragraphs could help.

File: src/test_worldgen.py [hunk 3]
```diff
@@ -15,6 +15,9 @@
     print(f'Ruled by: {scenario.ruler.title} {scenario.ruler.name}')
     print(scenario.kingdom.description, '\n')
     print(scenario.ruler.governance_style, scenario.ruler.evil_deeds)
+    for region in scenario.kingdom.regions:
```
Comment (bensku) [inline src/test_worldgen.py:18]: This is fine for test code, but the world generation we're demoing probably shouldn't include this, to leave some surprises to players :)

**Review by bensku — APPROVED** [review]
This was a bit large to review by just reading the code, but looks good!

File: src/Generation/Descriptions/area/init/model.py (hunk 1 above)
Comment (cs-szabolcs) [reply src/Generation/Descriptions/area/init/model.py:3]: Yeah, but by generating them this close together, it feels like the LLM pays more attention to the mission goal, and the areas are moving towards that goal both by names and by description.

These methods could also work if we return the previously generated areas to the LLM. This one just seemed simpler and works always, at most, the 2nd time.

File: src/Generation/Descriptions/summarize_context/model.py (hunk 2 above)
Comment (cs-szabolcs) [reply src/Generation/Descriptions/summarize_context/model.py]: Oh, I didn't know that. Acutally, I haven't test it much, I just observed that this way the result is shorter than without it. But I'll rewrite it to words then.

File: src/test_worldgen.py (hunk 3 above)
Comment (cs-szabolcs) [reply src/test_worldgen.py:18]: I thought this is just for test, not part of the demo

**Review by mfk99 — APPROVED** [review]

**Pull request merged**